### PR TITLE
Docs: Add best practices

### PR DESCRIPTION
**Pull request by oddlittlebird**
This PR adds a Best practices section to the documentation. I hope it will encourage people to contribute their ideas for more best practices in the future.

Edits and suggestions are encouraged!

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/best-practices/best-practices-for-creating-dashboards.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+### Write it down
+
+Once you have a strategy or design guidelines, write them down to help maintain consistency over time. Check out this [Wikimedia runbook example](https://wikitech.wikimedia.org/wiki/Performance/Runbook/Grafana_best_practices).
+
+## Best practices to follow
+
+- When creating a new dashboard, make sure it has a meaningful name.
+  - If you are creating a dashboard to play or experiment, then put the word `TEST` in the name.
+  - Consider including your name or initials in the dashboard name so that people know who owns the dashboard.
+  - Remove temporary experiment dashboards when you should remove are done with them.
```
Comment (peterholmberg) [inline docs/sources/best-practices/best-practices-for-creating-dashboards.md:46]:
```suggestion
  - Remove temporary experiment dashboards when you are done with them.
```

Comment (peterholmberg) [reply docs/sources/best-practices/best-practices-for-creating-dashboards.md:46]: Like this?

File: docs/sources/best-practices/best-practices-for-managing-dashboards.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+### Strategic observability
+
+There are several [common observability strategies]({{< relref "common-observability-strategies.md" >}}). You should research them and decide whether one of them works for you or if you want to come up with your own. Either way, have a plan, write it down, and stick to it. 
+
+Adapt your strategy to changing needs as necessary.
+
+### Maturity level
+
+What is your dashboard maturity level? Analyze your current dashboard setup and compare it to the [Dashboard management maturity levels]({{< relref "dashboard-management-maturity-levels.md" >}}). Understanding where you are can help you decide how to get where you want to be.
```
Comment (peterholmberg) [inline docs/sources/best-practices/best-practices-for-managing-dashboards.md:25]:
```suggestion
What is your dashboard maturity level? Analyze your current dashboard setup and compare it to the [Dashboard management maturity levels]({{< relref "dashboard-management-maturity-levels.md" >}}). Understanding where you are can help you decide how to get to where you want to be.
```
Maybe like this?

File: docs/sources/best-practices/dashboard-management-maturity-levels.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  {{< imgbox max-width="100%" img="/img/docs/best-practices/drill-down-example.png" caption="Example of using drill-down" >}}
+
+- Dashboard design reflects service hierarchies. The example shown below uses the RED method (request and error rate on the left, latency duration on the right) with one row per service. The row order reflects the data flow.
+
+  {{< imgbox max-width="100%" img="/img/docs/best-practices/service-hierarchy-example.png" caption="Example of a service hierarchy" >}}
+
+- Compare like to like: split service dashboards when the magnitude differs. Make sure aggregated metrics don't drown out important information.
+- Expressive charts with meaningful use of color and normalizing axes where you can. 
+  - Example of meaningful color: Green means it's good, red means it's bad. [Thresholds]({{< relref "../panels/thresholds.md" >}}) can help with that.
+  - Example of normalizing axes: When comparing CPU usage, measure by percentage rather than raw number, because each CPU might be a different size. This reduces cognitive load, because the user doesn't have to compute how much space is left on the CPU.
```
Comment (peterholmberg) [inline docs/sources/best-practices/dashboard-management-maturity-levels.md:52]: Is this about hard drive usage? CPU is usually measured by load in percent

**Review by peterholmberg — CHANGES_REQUESTED** [review]
Super nice work, learned a lot :slightly_smiling_face:. I think I found some words that are out of place?

File: docs/sources/best-practices/dashboard-management-maturity-levels.md (hunk 3 above)
Comment (oddlittlebird) [reply docs/sources/best-practices/dashboard-management-maturity-levels.md:52]: It's an example from @davkal's video for sleep-deprived devs. I was hoping he would review and be able to look these over...

**Review by peterholmberg — APPROVED** [review]

File: docs/sources/best-practices/dashboard-management-maturity-levels.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  {{< imgbox max-width="100%" img="/img/docs/best-practices/drill-down-example.png" caption="Example of using drill-down" >}}
+
+- Dashboard design reflects service hierarchies. The example shown below uses the RED method (request and error rate on the left, latency duration on the right) with one row per service. The row order reflects the data flow.
+
+  {{< imgbox max-width="100%" img="/img/docs/best-practices/service-hierarchy-example.png" caption="Example of a service hierarchy" >}}
+
+- Compare like to like: split service dashboards when the magnitude differs. Make sure aggregated metrics don't drown out important information.
+- Expressive charts with meaningful use of color and normalizing axes where you can. 
+  - Example of meaningful color: Green means it's good, red means it's bad. [Thresholds]({{< relref "../panels/thresholds.md" >}}) can help with that.
```
Comment (sakjur) [inline docs/sources/best-practices/dashboard-management-maturity-levels.md:51]:
```suggestion
  - Example of meaningful color: Blue means it's good, red means it's bad. [Thresholds]({{< relref "../panels/thresholds.md" >}}) can help with that.
```

Red/green color blindness is by far the most common form of color blindness. While green for good and red for bad is deeply embedded in our culture it also excludes tons of people, so it's a norm worth challenging when we're developing best practices.

File: docs/sources/best-practices/dashboard-management-maturity-levels.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  - Example of normalizing axes: When comparing CPU usage, measure by percentage rather than raw number, because each CPU might be a different size. This reduces cognitive load, because the user doesn't have to compute how much space is left on the CPU.
+- Directed browsing cuts down on "guessing."
+  - Template variables make it harder to “just browse” randomly or aimlessly.
+  - Most dashboards should be linked to by alerts.
+  - Browsing is directed with links. For more information, refer to [Linking]({{< relref "../linking/_index.md" >}}).
+- Version-controlled dashboard JSON.
+
+## High - optimized use
+
+At this stage, you have optimized your dashboard management use with a consistent and thoughtful strategy. It requires maintenance, but the results are worth it. Other IT departments want to grow up and be like you.
```
Comment (sakjur) [inline docs/sources/best-practices/dashboard-management-maturity-levels.md:61]:
```suggestion
At this stage, you have optimized your dashboard management use with a consistent and thoughtful strategy. It requires maintenance, but the results are worth it.
```

There's a lot of red tape involved in the optimized use best practices and consistency is naturally opposing experimentation / development. It's a higher level of sophistication in the same way as Marriott is on a higher level of sophistication than a family-owned inn.

Some people want that. Others don't. I think it's our role to inform about the different ways to think about dashboard management and let people make the judgment about what level of sophistication they'd want to put themselves on.

**Review by sakjur — COMMENTED** [review]
Some suggestions for consideration 🙂 I haven't seen the source material, so I've tried to explain my reasoning to make it easier for you accept or dismiss my reasoning.

File: docs/sources/best-practices/best-practices-for-creating-dashboards.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+## Best practices to follow
+
+- When creating a new dashboard, make sure it has a meaningful name.
+  - If you are creating a dashboard to play or experiment, then put the word `TEST` in the name.
+  - Consider including your name or initials in the dashboard name so that people know who owns the dashboard.
+  - Remove temporary experiment dashboards when you are done with them.
+- If you create many related dashboards, think about how to cross-reference them for easy navigation. Refer to [Best practices for managing dashboards]({{< relref "best-practices-for-managing-dashboards.md" >}}) for more information.
+- Grafana retrieves data from a data source. A basic understanding of [data sources]({{< relref "../features/datasources/data-sources.md" >}}) in general and your specific is important.
+- Avoid unnecessary dashboard refreshing to reduce the load on the network or backend. For example, if your data changes every hour, then you don't need to set the dashboard refresh rate to 30 seconds.
+- Use the left and right Y-axes when displaying time series with different ranges.
```
[the remark this answers is not in the document]
Comment (oddlittlebird) [reply docs/sources/best-practices/best-practices-for-creating-dashboards.md:50]: The source said ranges, and it had an example. I think units makes more sense.

File: docs/sources/best-practices/dashboard-management-maturity-levels.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  - Example of normalizing axes: When comparing CPU usage, measure by percentage rather than raw number, because each CPU might be a different size. This reduces cognitive load, because the user doesn't have to compute how much space is left on the CPU.
+- Directed browsing cuts down on "guessing."
+  - Template variables make it harder to “just browse” randomly or aimlessly.
+  - Most dashboards should be linked to by alerts.
+  - Browsing is directed with links. For more information, refer to [Linking]({{< relref "../linking/_index.md" >}}).
+- Version-controlled dashboard JSON.
+
+## High - optimized use
+
+At this stage, you have optimized your dashboard management use with a consistent and thoughtful strategy. It requires maintenance, but the results are worth it.
```
Comment (davkal) [inline docs/sources/best-practices/dashboard-management-maturity-levels.md:61]: At this level there should be a second Grafana instance that allows experimentation. When dashboards in this more ephemeral source prove to be permanently useful, then those dashboards should be added to the fully controlled/scripted Grafana instance.

**Review by davkal — APPROVED** [review]
Awesome writeup! Left a couple of comments.

File: docs/sources/best-practices/best-practices-for-creating-dashboards.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+### Write it down
+
+Once you have a strategy or design guidelines, write them down to help maintain consistency over time. Check out this [Wikimedia runbook example](https://wikitech.wikimedia.org/wiki/Performance/Runbook/Grafana_best_practices).
+
+## Best practices to follow
+
+- When creating a new dashboard, make sure it has a meaningful name.
+  - If you are creating a dashboard to play or experiment, then put the word `TEST` or `TMP` in the name.
+  - Consider including your name or initials in the dashboard name so that people know who owns the dashboard.
```
Comment (weeco) [inline docs/sources/best-practices/best-practices-for-creating-dashboards.md:45]: What about putting such information into the tags?

Comment (marcusolsson) [reply docs/sources/best-practices/best-practices-for-creating-dashboards.md:45]: I'm in favor of putting information like this as tags. `TEST` or `TMP` doesn't describe the purpose of the dashboard, but I'd like to be able to quickly filter them out. 

In the case of ownership, I've found that creating folder for individual team members can help here. Sometimes you just want to experiment with new dashboards, and giving people their own "sandbox" folder makes it clear who owns it. I also don't believe that you should link production dashboards to individual team members, i.e. personal folder should only ever contain experiments/tests/temporary dashboards. Once a dashboard is ready for prime time, it should be moved into the proper folder.

Comment (sakjur) [reply docs/sources/best-practices/best-practices-for-creating-dashboards.md:45]: I like this idea, and I know that we have at least some really big installations that does it that way. I still think, for sorting reasons, that prefixing with something that groups your dashboards together makes sense — but it makes sense to use tags for this at the moment.

For the future, if Grafana had personal/team dashboards and user profiles as first-class citizens that would solve the experiment/testing/temporary use-case (and potentially eliminate the need for setting up a second Grafana for the dashboards that make it into the mainline that @davkal mentioned in https://github.com/grafana/grafana/pull/27057#discussion_r476640826 ).

I couldn't find it now, but I've seen some discussion somewhere around having label-tags with key-value separation, which would be another way of adding something like `owner:emil` without having to clutter the tags UX with a lot of tags that are really drill downs from a general tag category. That would help removing some of the arguments against recommending tags as a best practice for say dashboard ownership.

Comment (oddlittlebird) [reply docs/sources/best-practices/best-practices-for-creating-dashboards.md:45]: That might work if there aren't too many users or dashboard creators.

**Comment by kogent** [comment]
fwiw, the dashboard maturity link is broken. the link goes to [model](https://grafana.com/docs/grafana/latest/best-practices/dashboard-management-maturity-model/) when it should go to [levels](https://grafana.com/docs/grafana/latest/best-practices/dashboard-management-maturity-levels/).  

There are also broken images on the working link.

**Comment by oddlittlebird** [comment]
Thank you!

**Comment by oddlittlebird** [comment]
Where are you seeing broken images?

**Pull request merged**